core/tests/webdriverio_utils/ExplorationPlayerPage.js

```diff
@@ -23,63 +23,260 @@ var action = require('./action.js');
   var feedbackTextArea = $('.e2e-test-exploration-feedback-textarea');
+  var flaggedSuccessElement = $(
+    '.e2e-test-exploration-flagged-success-message');
+  var gotItButton = $('.e2e-test-learner-got-it-button');
+  var infoCardRating = $('.e2e-test-info-card-rating');
+  var nextCardButton = $('.e2e-test-continue-to-next-card-button');
+  var pauseButton = $('.e2e-test-pause-circle');
+  var playButton = $('.e2e-test-play-circle');
   var ratingStar = $('.e2e-test-rating-star');
   var ratingStarsSelector = function() {
     return $$('.e2e-test-rating-star');
   };
+  var reportExplorationButton = $('.e2e-test-report-exploration-button');
+  var returnToParentButton = $('.e2e-test-return-to-parent-button');
+  let submitButton = $('.e2e-test-submit-report-button');
+  var suggestionPopupLink = $('.e2e-test-exploration-suggestion-popup-link');
+  var viewHintButton = $('.e2e-test-view-hint');
+  var viewSolutionButton = $('.e2e-test-view-solution');
+  var voiceoverLanguageSelector = $('.e2e-test-audio-lang-select');
   var waitingForResponseElem = $('.e2e-test-input-response-loading-dots');
 
+  this.expandAudioBar = async function() {
+    await action.click('Audio Bar Expand Button', audioBarExpandButton);
+  };
+
+  this.pressPlayButton = async function() {
+    await action.click('Play Button', playButton);
+  };
+
+  this.expectAudioToBePlaying = async function() {
+    await waitFor.visibilityOf(
+      pauseButton, 'Pause button taking too long to show up.');
+  };
+
+  this.pressPauseButton = async function() {
+    await action.click('Pause Button', pauseButton);
+  };
+
+  this.expectAudioToBePaused = async function() {
+    await waitFor.visibilityOf(
+      playButton, 'Play button taking too long to show up.');
+  };
+
+  this.changeVoiceoverLanguage = async function(language) {
+    await waitFor.visibilityOf(
+      voiceoverLanguageSelector, 'Language selector takes too long to appear.');
+    await voiceoverLanguageSelector.selectByVisibleText(language);
+  };
+
+  this.clickThroughToNextCard = async function() {
+    await waitFor.elementToBeClickable(nextCardButton);
+    await action.click('Next Card button', nextCardButton);
+  };
+
+  this.clickSuggestChangesButton = async function() {
+    await action.click('Suggestion Popup link', suggestionPopupLink);
+  };
+
+  this.expectNextCardButtonTextToBe = async function(text) {
+    await waitFor.visibilityOf(
+      nextCardButton, 'Next Card Button not showing up.');
+    var buttonText = await action.getText('Next Card Button', nextCardButton);
+    expect(buttonText).toMatch(text);
+  };
+
+  this.reportExploration = async function() {
+    await action.click('Report Exploration Button', reportExplorationButton);
+    let radioButton = await $$('<input>')[0];
+    await waitFor.visibilityOf(
+      radioButton, 'Radio Buttons takes too long to appear');
+    await action.click('Radio Button', radioButton);
+    // eslint-disable-next-line oppia/e2e-practices
```

Comment:
Why are we doing this? We shouldn't be bypassing the lint checks.

Reply:
We have lint check for not letting the user use any other class name other than e2e-test-* . In protractor for selecting a element using a class name we have `element(by.css())` , so we have added the pattern in lint check to find by.css and then throw error if e2e-test-* classname is not present. But the problem in webdriverio is we use `$()` for each selection method, like for class name `$('.class-name')`, for tagname `$('<tag-name>')` etc, there is no diference in pattern and hence our lint check throw error even if the selction is valid as it only looks for pattern $('') and throws error if .e2e-classname-* is not present.

Reply:
Then why don't we give the textarea a classname too?

Either we should modify the lint check or we should follow it throughout, I think.

Reply:
Done. Added the classname to textArea.